src/components/designSystem/Table/Table.tsx

```diff
@@ -216,38 +436,51 @@ export const Table = <T extends DataItem>({
   return (
     // Width is set to 0 and minWidth to 100% to prevent table from overflowing its container
     // cf. https://stackoverflow.com/a/73091777
-    <Box
-      width={0}
-      minWidth="100%"
-      overflow="auto"
-      height="100%"
-      sx={{
-        transform: 'translateZ(0)',
-      }}
-    >
-      <StyledTable
+    <div className="h-full w-0 min-w-full overflow-auto">
+      <MUITable
+        className="border-collapse"
         data-test={TABLE_ID}
         id={TABLE_ID}
         ref={tableRef}
-        $containerSize={containerSize}
-        $rowSize={rowSize}
+        sx={{
+          '& .lago-table-cell:first-of-type .lago-table-inner-cell': {
+            ...setResponsiveProperty('paddingLeft', containerSize),
+          },
+          '& .lago-table-cell:last-of-type .lago-table-inner-cell': {
+            ...setResponsiveProperty('paddingRight', containerSize),
+          },
+          '& .lago-table-inner-cell': {
+            minHeight: rowSize,
+          },
+        }}
       >
-        <TableHead>
+        <MUITableHead>
           <tr>
             <>
               {filteredColumns.map((column, i) => (
                 <TableCell
+                  className="sticky top-0 z-[1] border-b-0 bg-white shadow-b"
```

Comment:
I would have gone with `z-10` directly to not enforce arbitrary value usage

Reply:
Yeah indeed, and I don't think it matters that much to have 1 instead of 10, changing!